### ISSUE TITLE
[BUG] Intermittant error about removing a file on rust-nightly

### ISSUE THREAD

**Issue by mattgodbolt** [compiler-explorer/compiler-explorer#1565]
**Describe the bug**
There's an issue where we intermittantly see 

```
error: failed to remove ./output.example.3a1fbbbh-cgu.0.rcgu.s: No such file or directory (os error 2)
```

**To Reproduce**

* https://rust.godbolt.org/z/YdIs_b

reload a few times

**Expected behavior**
No error

**Additional context**
I run the code in a loop on the admin node and it was fine. Make me wonder if it's a problem with the isolation. #1440 may "fix" (or at least change this...)

**Comment by mattgodbolt** [comment]
Confirmed it's still hapenning

**Comment by OfekShilon** [comment]
Probably solved by #5988. Awaiting deployment to test.
